receiver/pprofreceiver/convert.go

```diff
@@ -0,0 +1,415 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package pprofreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/pprofreceiver"
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+
+	"github.com/google/pprof/profile"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pprofile"
+	semconv "go.opentelemetry.io/otel/semconv/v1.37.0"
+)
+
+// errInvalPprof is returned for invalid pprof data.
+var errInvalPprof = errors.New("invalid pprof data")
+
+const (
+	// noAttrUnit is an internal helper to indicate that no
+	// unit is associated to this Attribute.
+	noAttrUnit = int32(-1)
+)
+
+// attr is a helper struct to build pprofile.ProfilesDictionary.attribute_table.
+type attr struct {
+	keyStrIdx  int32
+	value      any
+	unitStrIdx int32
+}
+
+// fn is a helper struct to build pprofile.ProfilesDictionary.function_table.
+type fn struct {
+	name       string
+	systemName string
+	fileName   string
+	startLine  int64
+}
+
+// lookupTables is a helper struct around pprofile.ProfilesDictionary.
+type lookupTables struct {
+	stringTable        map[string]int32
+	lastStringTableIdx int32
+
+	attributeTable        map[attr]int32
+	lastAttributeTableIdx int32
+
+	functionTable        map[fn]int32
+	lastFunctionTableIdx int32
+}
+
+func convertPprofToPprofile(src *profile.Profile) (*pprofile.Profiles, error) {
```

Comment:
Not a blocker for this PR, but if we are willing to add a pprof exporter, we could consider moving this to a [pprof translator package](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/translator)